scvi/dataset/dataset.py

```diff
@@ -1468,6 +1476,105 @@ def get_batch_mask_cell_measurement(self, attribute_name: str):
         return batch_mask
 
 
+def poisson_gene_selection(
+    X,
+    n_top_genes: int = 4000,
+    use_cuda = True,
+    n_samples: int = 10000,
+    silent : bool = False,
+    batch_size : int = 5000,
+    **kwargs
+):
+    """ Rank and select genes based on the enrichment of zero counts in data
+        compared to a Poisson count model.
+
+        This is based on M3Drop: https://github.com/tallulandrews/M3Drop
+
+        The method accounts for library size internally, a raw count matrix should be provided.
+
+        Instead of Z-test, enrichment of zeros is quantified by posterior
+        probabilites from a binomial model, computed through sampling.
+
+        :param X: Count matrix, preferably a sparse matrix.
+        :param n_top_genes: How many variable genes to select.
+        :param use_cuda: Default: ``False``.
+        :param n_samples: The number of Binomial samples to use to estimate posterior probability
+            of enrichment of zeros for each gene. Default: ``10000``.
+        :param silent: If ``True``, disables the progress bar. Default ``False``.
+        :param batch_size: Size of temporary matrix for incremental calculation. Larger
+            is faster but requires more RAM or GPU memory. (The default should be fine unless
+            there are hundreds of millions cells or millions of genes.) Default ``5000``.
+
+        :return: A ``pd.DataFrame`` with per-gene statistics.
+
+    """
+    from tqdm import tqdm
```

Comment:
btw you can import tqdm like in trainer.py if you want it to adapt to jupyter notebooks (unless you like the simpler look). In trainer we also direct it to stdout to avoid the red stuff in jupyter.

Reply:
Oh that's how you get rid of the red stuff! Will change that. Yeah I prefer the text output because it doesn't go away when you re-open the notebook.. It always looks like something is wrong with the message about a missing javascript thing.

Reply:
Yeah I think we should change it back, I agree it's a bit more annoying than good